pkg/tests/tasks/observability/yaml/federated-monitor.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: istio-federation
+spec:
+  selector:
+    matchLabels:
+      app: prometheus
+  endpoints:
+  - targetPort: 9090
+    path: /federate
+    interval: 30s
+    scrapeTimeout: 30s
+    honorLabels: true
+    params:
+      'match[]':
+        - '{job="kubernetes-pods"}'
+        - '{job="kubernetes-services"}'
+    relabelings:
+    - action: replace
+      replacement: "unique-mesh-id"
+      targetLabel: mesh_id
```

Comment:
You don't use this label in the test, so you can remove it or add a check for this label. I would prefer the second option.